### PR TITLE
Write some unit tests

### PR DESCRIPTION
**Pull request by Patryk-Kuchta**
Write some unit tests

Your goal is write unit tests for league.js.

Look at league.js. The exports.createLeague function returns an object with four methods:

    addPlayer

    getPlayers

    recordWin

    getWinner

Think about how you would test each of these, and write tests for them using league_test.js as a starting point.

Ignore the other test files for now - we'll be using those in future exercises.

Remember to think about edge cases as well as regular input
Hints

    You can run a single test or group of tests by

        changing it to it.only(...) or describe.only(...)

        or by using the mocha target with the grep style selector -g: npm run mocha -- -g "#addPlayer". (using quotes to include # in pattern)

    The test file imports the Chai assertion library, which gives you lots of options for checking whether values are correct. If you find yourself writing complicated assertions, look at the Chai documentation to see if it can help.

## Summary by CodeRabbit

- **Tests**
	- Introduced new test cases for the league module to enhance coverage.
	- Added tests for player addition validation, ensuring invalid names are correctly rejected.
	- Expanded tests for retrieving players and validating their structure.
	- Enhanced win recording tests to enforce game logic and input conditions.
	- Added tests for determining the winner and handling empty league scenarios.
	- Implemented a helper function to streamline league setup for tests.

### CODE REVIEW

File: test/league_test.js [hunk 1]
```diff
@@ -14,5 +14,103 @@ describe('league', function () {
...
+      league.recordWin('Bob', 'Amy');
+
+      expect(league.getWinner()).to.equal('Bob');
+    });
+
+    it('yields null for an empty league', function () {
+      const league = gameState.createLeague();
+
+      expect(league.getWinner()).to.be.null;
+    });
```
[the remark this answers is not in the document]
Comment (Patryk-Kuchta) [reply test/league_test.js]: I like the test case you added, but I dislike that you are not reusing the same set of names as in the original, please fix
